index.js

```diff
@@ -69,24 +70,33 @@ function processFiles(fileNames, config) {
 	return Promise.all(fileNames.map(fileName => processFile(fileName, config)));
 }
 
-function processFile(fileName,
+async function processFile(fileName,
 		{ source, target, targetDir, fingerprint, assetManager, plugins }) {
 	let sourcePath = path.join(source, fileName);
 	let targetPath = path.join(target, fileName);
 
-	return readFile(sourcePath).
-		then(content => {
-			let type = determineFileType(sourcePath);
-			let plugin = type && plugins[type];
-			return plugin ? plugin(content) : content;
-		}).
-		then(content => {
-			let options = { targetDir };
-			if(fingerprint !== undefined) {
-				options.fingerprint = fingerprint;
-			}
-			return assetManager.writeFile(targetPath, content, options);
-		});
+	try {
+		var content = await readFile(sourcePath); // eslint-disable-line no-var
+	} catch(err) {
+		if(err.code !== "ENOENT") {
+			throw err;
+		}
+		console.error(`WARNING: \`${sourcePath}\` no longer exists`);
+		return;
```

Comment:
can we safely return `undefined` here?

Reply:
From my understanding: yes.

Reply:
I'm not so sure, tbh - but I'm still struggling to fully grok the data flow here, at least without expending more energy than I can muster at the moment.